src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts

```diff
@@ -952,7 +954,7 @@ export class AnythingQuickAccessProvider extends PickerQuickAccessProvider<IAnyt
 			icon = resourceOrEditor.getIcon();
 		} else {
 			resource = URI.isUri(resourceOrEditor) ? resourceOrEditor : resourceOrEditor.resource;
-			label = basenameOrAuthority(resource);
+			label = this.customEditorLabelService.getName(resource) || basenameOrAuthority(resource);
```

Comment:
I am not entirely sure this is a good location to compute the custom editor label: this method will be called not only for recently opened editors, but also all search results. This leads to quite some inconsistency:
* you get the custom label for file results but you cannot search for it
* you see custom labels for file resources but only here, not in any other part of the UI (e.g. explorer)

I would suggest to limit the use of custom editor labels to entries in the picker that are "recently opened" editors.

![Screenshot 2024-04-07 at 19 43 32](https://github.com/microsoft/vscode/assets/900690/ebb4f79a-e724-4477-adf8-9e12d3edf3ff)
![Screenshot 2024-04-07 at 19 43 07](https://github.com/microsoft/vscode/assets/900690/108e6689-605d-4f8b-8618-714d19bc39fd)

Reply:
I think it makes more sense to show it for all and not only `recently opened`. We will need to figure out where it makes sense to add the custom labels and where not, but for me it makes sense to have it here. Being able to search for them makes sense to me, but I don't think we can allow it from a performance perspective (we could try and measure but let's not do that now)

Reply:
We could consider appending the file name to the description when there is a custom label present. This would allow the user to see both the real and custom names